dashboards-observability/public/components/application_analytics/components/application.tsx

```diff
@@ -0,0 +1,244 @@
+import { RAW_QUERY } from '../../../../common/constants/explorer';
+import { NotificationsStart } from '../../../../../../src/core/public';
+import { AppAnalyticsComponentDeps } from '../home';
+
+
+const TAB_OVERVIEW_ID = uniqueId(TAB_OVERVIEW_ID_TXT_PFX);
+const TAB_SERVICE_ID = uniqueId(TAB_SERVICE_ID_TXT_PFX);
+const TAB_TRACE_ID = uniqueId(TAB_TRACE_ID_TXT_PFX);
+const TAB_LOG_ID = uniqueId(TAB_LOG_ID_TXT_PFX);
+const TAB_CONFIG_ID = uniqueId(TAB_CONFIG_ID_TXT_PFX);
+
+export interface DetailTab {
+  id: string;
+  label: string;
+  description: string;
+  onClick: () => void;
+  testId: string;
+}
+
+interface AppDetailProps extends AppAnalyticsComponentDeps {
+  disabled?: boolean;
+  appId: string;
+  pplService: PPLService;
+  dslService: DSLService;
+  savedObjects: SavedObjects;
+  timestampUtils: TimestampUtils;
+  notifications: NotificationsStart;
+}
+
+
+export function Application(props: AppDetailProps) {
+  const { pplService, dslService, timestampUtils, savedObjects, http, notifications } = props;
+  const [selectedTabId, setSelectedTab] = useState<string>(TAB_OVERVIEW_ID);
+  const handleContentTabClick = (selectedTab: IQueryTab) => setSelectedTab(selectedTab.id);
+  const history = useHistory();
+  const [toasts, setToasts] = useState<Array<Toast>>([]);
+
+  const setToast = (title: string, color = 'success', text?: ReactChild, side?: string) => {
+    if (!text) text = '';
+    setToasts([...toasts, { id: new Date().toISOString(), title, text, color } as Toast]);
+  };
+
+  const getExistingEmptyTab = ({tabIds, queries, explorerData}: EmptyTabParams) => {
+    let emptyTabId = '';
+    for (let i = 0; i < tabIds!.length; i++) {
+      const tid = tabIds![i];
+      if (isEmpty(queries[tid][RAW_QUERY]) && isEmpty(explorerData[tid])) {
+        emptyTabId = tid;
+        break;
+      }
+    }
+    return emptyTabId;
+  };
+
+
+  const getOverview = () => {
+    return (
+      <Dashboard {...props} page="app" appId="id" appName="Cool Application" />
+    );
+  };
+
+  const getService = () => {
+    return (
+      <Services {...props} page="app" appId="id" appName="Cool Application" />
+    );
+  };
+
+  const getTrace = () => {
+    return (
+      <>
+      <Traces {...props} page="app" appId="id" appName="Cool Application" />
+      <EuiSpacer size='m'/>
+      <SpanDetailPanel
+        {...props}
+        traceId="id"
+        colorMap="color" 
+      />
+    </>
+    );
+  };
+
+  const getLog = () => {
+    return (
+      <LogExplorer 
+        pplService={pplService}
+        dslService={dslService}
+        savedObjects={savedObjects}
+        timestampUtils={timestampUtils}
+        http={http} 
+        history={history} 
+        notifications={notifications} 
+        setToast={setToast}
+        // App Analytics will not be saving queries on Log Events
+        savedObjectId={''} 
+        getExistingEmptyTab={getExistingEmptyTab}
+        />
```

Comment:
Minor: Also indent issue here and some other places.

Reply:
Will fix in next PR